File: 001-doors.js
// solution of the famous "100 doors problem" from freeCodeCamp
function getFinalOpenedDoors() {
    let doors = Array(100).fill(false);
    //[false , false , false ,...false]
    for (let i = 0; i < doors.length ; i++){
        for (let j = 0; j < doors.length ; j++){
            if ((i+1) % (j+1) === 0) {
                doors[i] = !doors[i];
            }
        }
    }
    return doors.reduce((acc , state , index) => {
        if(state === true){
            acc.push(index + 1);
        }
        return acc;
    }, []);
}

File: 001-doors.py
# solution of the "100 doors" chalenge from the rosetta code section of freeCodeCamp
def toggle_doors(num_doors):
    # Create a list of doors, initially all closed (False)
    doors = [False] * num_doors

    # Toggle the state of the doors according to the rules
    for i in range(1, num_doors + 1):
        for j in range(i - 1, num_doors, i):
            doors[j] = not doors[j]

    # Collect the numbers of the open doors
    open_doors = [index + 1 for index, state in enumerate(doors) if state]
    
    return open_doors
print(toggle_doors(100))
# Test